### PR TITLE
feat(Core/Movement): Allow waypoints to use 0 as valid facing value

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/4747515872775553de35cc3267e1ccd5e0076ca4)

Co-Authored-By: Ovah <18347559+Ovahlord@users.noreply.github.com>

## Changes Proposed:
-  make waypoint orientation values nullable to support angle values of 0 as well as there actually are some cases that use such values
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- not tested
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]

File: src/server/game/AI/SmartScripts/SmartAI.cpp [hunk 1]
```diff
@@ -257,9 +257,10 @@ void SmartAI::PausePath(uint32 delay, bool forced)
         me->GetMotionMaster()->MoveIdle();//force stop
 
         auto waypoint = mWayPoints->find(mCurrentWPID);
-        if (float orientation = waypoint->second->o)
+        Optional<float> orientation = waypoint->second->o;
+        if (orientation)
```
Comment (99dsimonp) [inline src/server/game/AI/SmartScripts/SmartAI.cpp:261]: Consider dropping variable `orientation`: 
`if(waypoint->second->o)`

**Comment by acidmanifesto** [comment]
Can we get someone to test this in the dk starter zone? I think the 3rd mission or so with the you taking on a unworth dk initiative they always have their back turned until the duel count down is finished and the fight stops.

**Comment by acidmanifesto** [comment]
@Kitzunu is this still considered wip? It tested all good on my end.

**Comment by Kitzunu** [comment]
Then it is good to go

**Pull request merged**